### PR TITLE
Read stderr from server for a short time before exiting

### PR DESCRIPTION
**Pull request by havocp**
Previously we exited immediately without waiting to get any
error output.

This patch also adds the command line and directory to the
exception message on failure, in case the failure is due to
getting one of those wrong, for example.

### CODE REVIEW

File: launch/src/main/scala/xsbt/boot/ServerApplication.scala [hunk 1]
```diff
@@ -107,7 +107,19 @@ class StreamDumper(in: java.io.BufferedReader, out: java.io.PrintStream) extends
...
-  def close(): Unit = running.set(false)
+  def close(waitForErrors: Boolean): Unit = {
+    // closing "in" blocks forever on Windows, so don't do it;
+    // just wait a couple seconds to read more stuff if there is
+    // any stuff.
+    if (waitForErrors) {
+      endTime.set(System.currentTimeMillis + 2000)
+      // let ourselves read more (thread should exit on earlier of endTime or EOF)
+      while (isAlive() && (endTime.get > System.currentTimeMillis))
+        Thread.sleep(50)
```
Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala:121]: Maybe a backoff of Thread.yield in case the endTime is <50 millis from now?

**Comment by jsuereth** [comment]
Nice!  I like it, one minor comment then good to merge.

File: launch/src/main/scala/xsbt/boot/ServerApplication.scala [hunk 2]
```diff
@@ -107,7 +107,22 @@ class StreamDumper(in: java.io.BufferedReader, out: java.io.PrintStream) extends
...
+    // just wait a couple seconds to read more stuff if there is
+    // any stuff.
+    if (waitForErrors) {
+      endTime.set(System.currentTimeMillis + 2000)
+      // at this point we'd rather the dumper thread run
+      // before we check whether to sleep
+      Thread.yield()
+      // let ourselves read more (thread should exit on earlier of endTime or EOF)
+      while (isAlive() && (endTime.get > System.currentTimeMillis))
+        Thread.sleep(50)
```
Comment (jsuereth) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala:121]: This is where I meant having the incremental backoff, but the above is pretty good.

Comment (havocp) [inline launch/src/main/scala/xsbt/boot/ServerApplication.scala:121]: I think I don't know what you mean...

**Comment by jsuereth** [comment]
@havocp Not compiling.  You need :

``` scala
Thread.`yield`()
```

**Comment by jsuereth** [comment]
Remaining failure is spurious cache issue.

**Pull request merged**